virt/virt.py

```diff
@@ -77,6 +77,37 @@ def toDict(self):
         return d
 
 
+class Hypervisor(object):
+    """
+    A model for information about a hypervisor
+    """
+    def __init__(self, hypervisorId,  guestIds=[], name=None):
+        """
+        Create a new Hypervisor that will be sent to subscription manager
+
+        'hypervisorId': the unique identifier for this hypervisor
+
+        'guestIds': a list of Guests
+
+        'name': the hostname, if available
+        """
+        self.hypervisorId = hypervisorId
+        self.guestIds = guestIds
+        self.name = name
+
+    def toDict(self):
+        d = {
+            'hypervisorId': self.hypervisorId,
```

Comment:
The hypervisor class is used for both but in the manager/subscriptionmanager.py Subscription Manager class the data to be sent is changed to the old format if necessary.

Re: the missing layer, I'll go change that :+1: